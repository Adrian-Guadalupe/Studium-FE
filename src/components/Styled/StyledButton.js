import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { Button } from 'antd';

const StyledButton = ({
  text,
  type = 'primary',
  size = 'default',
  icon,
  shape,
  loading,
  block,
  color, 
  position = "none",
  top,
  left,
  ...props
}) => {
  return (
    <StyledAntdButton
      type={type}
      size={size}
      icon={icon}
      shape={shape}
      loading={loading && 'loading'}
      block={block && 'block'}      
      
      {...props}
      position={position}
      top = {top}
      left={left}
    >
      {text}
    </StyledAntdButton>
  );
};

StyledButton.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf(['primary', 'dashed', 'danger', 'link', 'darkgray']),
  size: PropTypes.oneOf(['large', 'default', 'small']),
  icon: PropTypes.string,
  shape: PropTypes.oneOf(['circle', 'round']),
  loading: PropTypes.bool,
  block: PropTypes.bool,
};

const StyledAntdButton = styled(Button)`
  && {
    border: 0;
    border-radius:15px;
    color: ${props => {
      if(props.type === 'darkgray') {
        return 'white'
      } else {
        return 'black'
      }
    }};
    
    background-color: ${props =>{
     if( props.type === 'primary'){
       return props.theme.mainColor
     } else if (props.type === 'darkgray') {
      return props.theme.darkGray
     } else { 
       return props.theme.mainColor
     }
  }};
  position: ${props => {
    if(props.position !== "none"){
      return props.position
    }
    return ""
  }};
  top: ${props => {
    if(props.position !== "none"){
      return props.top
    }
    return ""

    }};
    left: ${props => {
      if(props.position !== "none"){
        return props.left
    }
    return ""
    }};

    width: ${props => {
      if(props.size === "large"){
        return "204px"
    }
    return ""
    }};

    height: ${props => {
      if(props.size === "large"){
        return "62px"
    }
    return ""
    }};
}
`

export default StyledButton;
