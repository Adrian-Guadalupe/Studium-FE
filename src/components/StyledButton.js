import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { Button } from "antd";

const StyledButton = ( { text, type = "primary", size = "default", icon, shape, loading, block, ...props } ) => {
  return ( <StyledAntdButton type={ type } size={ size } icon={ icon }
                             shape={ shape }
                             loading={ loading && "loading" }
                             block={ block && "block" }{ ...props }
  >
    { text }
  </StyledAntdButton> );
};

StyledButton.propTypes = {
  text: PropTypes.string,
  type: PropTypes.oneOf( [ "primary", "dashed", "danger", "link" ] ),
  size: PropTypes.oneOf( [ "large", "default", "small" ] ),
  icon: PropTypes.string,
  shape: PropTypes.oneOf( [ "circle", "round" ] ),
  loading: PropTypes.bool,
  block: PropTypes.bool
};

const StyledAntdButton = styled( Button )`
  &&{
    background-color: ${ props => props.type == "danger" ? "red" :
  props.theme.mainColor };
  }
`;

export default StyledButton;