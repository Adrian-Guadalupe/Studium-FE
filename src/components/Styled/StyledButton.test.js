import React from 'react';
import styled from 'styled-components';
import testRenderer from 'react-test-renderer';
// import 'jest-styled-components';

import StyledButton from './StyledButton';


test('Button renderer correctly', () => {
    const tree = testRenderer.create(<StyledButton />).toJSON();
    expect(tree).toMatchSnapshot();
});

