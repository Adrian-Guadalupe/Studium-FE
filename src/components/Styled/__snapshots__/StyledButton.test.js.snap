// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renderer correctly 1`] = `
.c0.c0 {
  border: 0;
  color: black;
}

<button
  className="ant-btn c0 ant-btn-primary"
  onClick={[Function]}
  type="button"
/>
`;
